### PR TITLE
Adding  protein_folding widget.

### PR DESCRIPTION
**Pull request by Narsil**
output

The goal is to add a few proteins already ran by @Rocketknight1 , and for unknown proteins, instead of running any inference, we would redirect to a voting website for next proteins to be run.

https://poll.ly/#/PwjxBxj2 for instance.

### CODE REVIEW

File: widgets/package.json [hunk 1]
```diff
@@ -28,5 +28,8 @@
 		"tslib": "^2.0.0",
 		"typescript": "^4.0.0"
 	},
-	"type": "module"
+	"type": "module",
+	"dependencies": {
+		"ngl": "^0.10.4"
```
Comment (julien-c) [inline widgets/package.json:33]: what's the bundle size when adding this client side?

File: widgets/src/lib/InferenceWidget/shared/WidgetProteinViewer/WidgetProteinViewer.svelte [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	export let src: string;
+	export let pdb_src: string = "";
+	// create a `stage` object
+
+	onMount(async () => {
+		const NGL = await import("ngl");
+		var stage = new NGL.Stage("viewport");
+		// load a PDB structure and consume the returned `Promise`
+		pdb_src = src;
+		stage.loadFile(src).then(function (component) {
```
Comment (julien-c) [inline widgets/src/lib/InferenceWidget/shared/WidgetProteinViewer/WidgetProteinViewer.svelte:14]: you can probably flatten this `then` into an await

File: widgets/src/lib/InferenceWidget/widgets/ProteinFoldingWidget/ProteinFoldingWidget.svelte [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+				onClick={() => {
+					getOutput();
+				}}
+			/>
+		</form>
+	</svelte:fragment>
+	<svelte:fragment slot="bottom">
+		<WidgetProteinViewer
+			classNames="mt-4"
+			src="http://localhost:8000/ranked_0.pdb"
```
Comment (julien-c) [inline widgets/src/lib/InferenceWidget/widgets/ProteinFoldingWidget/ProteinFoldingWidget.svelte:138]: src seems bogus

File: widgets/package.json (hunk 1 above)
Comment (julien-c) [reply widgets/package.json:33]: according to their repo, >1MB so i don't think we would want to embed that on every page

**Review by julien-c — COMMENTED** [review]
I think this might fit better inside a Space, no? (I think using Gradio has a way to inject completely custom HTML/JS)

File: widgets/src/lib/InferenceWidget/widgets/ProteinFoldingWidget/ProteinFoldingWidget.svelte (hunk 3 above)
Comment (Narsil) [reply widgets/src/lib/InferenceWidget/widgets/ProteinFoldingWidget/ProteinFoldingWidget.svelte:138]: It's a way to test locally without having to use the real API.

File: widgets/package.json (hunk 1 above)
Comment (Narsil) [reply widgets/package.json:33]: How can we make it separate ?

Isn't the svelte compiler able to make such decisions already based on `import` statements ?

**Comment by Narsil** [comment]
> I think this might fit better inside a Space, no? (I think using Gradio has a way to inject completely custom HTML/JS)

Hm I don't think it's simple to add JS within GRADIO, as I have no way to know the names of the html tags.

I could dump raw html with gradio (I mean everything), but I am unsure of the benefits there. hosting simple html page is probably better (lazy JS loading etc..)

Here the benefit is that we can reuse the radio buttons to suggest existing molecules to user (which we would put in cache) so that we could keep the experience of instant results.

For unknown molecules the idea is to refer them to another place (voting website for instance) where maybe they could share the most interesting ones we should run (which we would in TURN add to the cache).

**Comment by Narsil** [comment]
Dropping this (stale)

**Comment by julien-c** [comment]
Note @Narsil that we now have a way for power users to just drop a `static` webpage in a Space now (and HF users can drop any JS they want inside it)

cc @cbensimon and @osanseviero

**Comment by Narsil** [comment]
Good to know, I'll drop it anyway, because of the obvious hard work to make everything work in inference + the open sourcing of many molecules from deepmind:

https://www.theguardian.com/technology/2021/jul/22/ai-firm-deepmind-puts-database-building-blocks-life-human-protein-structure-free-online

Still could be interesting for demo purposes when more appropriate. At least prep work exists.

**Comment by julien-c** [comment]
yes my note was more general, for future similar integrations

**Pull request closed without merging**